edward/inferences/variational_inference.py

```diff
@@ -123,6 +123,10 @@ def update(self, feed_dict=None):
 
     sess = get_session()
     _, t, loss = sess.run([self.train, self.increment_t, self.loss], feed_dict)
+    if self.logging and self.n_print != 0:
+      if t == 1 or t % self.n_print == 0:
+          summary = sess.run(self.summarize, feed_dict)
```

Comment:
So I think the issue here is that with this construction, you'd actually need
```python
if self.logging and self.n_print != 0:
      if t == 1 or t % self.n_print == 0:
          _, t, loss, summary = sess.run([self.train, self.increment_t, self.loss, self.summarize], feed_dict)
      else:
          _, t, loss, summary = sess.run([self.train, self.increment_t, self.loss], feed_dict)
else:
       _, t, loss = sess.run([self.train, self.increment_t, self.loss], feed_dict)
```
to cover the fact that you only want to summarize on the first iteration and every print iteration. My thinking was just that it seems like a lot of repetition.